File: sgf-poker/sgf-poker-ios/Core/UseCases/DeletePlayerUseCase.swift
// Core/UseCases/DeletePlayerUseCase.swift

import Foundation

/// Removes a player from the repository by their identifier.
final class DeletePlayerUseCase {

    private let repository: PlayerRepository

    init(repository: PlayerRepository) {
        self.repository = repository
    }

    /// - Parameter id: The UUID of the player to remove.
    /// - Throws: `PlayerError.playerNotFound` if no matching player exists.
    func execute(id: UUID) throws {
        try repository.delete(id: id)
    }
}

File: sgf-poker-android/app/src/main/java/com/sgf/poker/usecases/player/DeletePlayerUseCase.java
package com.sgf.poker.usecases.player;

import com.sgf.poker.data.repository.PlayerRepository;

/** Removes a player by id. */
public class DeletePlayerUseCase {

    private final PlayerRepository repository;

    public DeletePlayerUseCase(PlayerRepository repository) {
        this.repository = repository;
    }

    public void execute(String playerId) {
        repository.delete(playerId);
    }
}
